datafusion/physical-expr/src/expressions/datetime.rs

```diff
@@ -153,7 +154,10 @@ pub fn evaluate_array(
 ) -> Result<ColumnarValue> {
     let ret = match array.data_type() {
         DataType::Date32 => {
-            let array = array.as_any().downcast_ref::<Date32Array>().unwrap();
+            let array = match as_date32_array(&array) {
+                Ok(array) => array,
+                Err(e) => return Err(e),
+            };
```

Comment:
I think this can be like

```suggestion
            let array = as_date32_array(&array)?;
```

Reply:
The same comment applies to the rest of the PR